core/src/main/java/org/mobilitydata/gtfsvalidator/input/GtfsZipFileInput.java

```diff
@@ -32,28 +33,43 @@
 public class GtfsZipFileInput extends GtfsInput {
   private final ImmutableSet<String> filenames;
   private final ZipFile zipFile;
+  private boolean isMacZip;
 
-  public GtfsZipFileInput(ZipFile zipFile) {
+  public GtfsZipFileInput(ZipFile zipFile, String zipFileName) {
     this.zipFile = zipFile;
-
+    StringBuilder strBuilder = new StringBuilder();
+    strBuilder.append(zipFileName);
+    strBuilder.append("/");
+    String macDirectory = strBuilder.toString();
     ImmutableSet.Builder<String> filenamesBuilder = new ImmutableSet.Builder<>();
     for (Enumeration<ZipArchiveEntry> entries = zipFile.getEntries(); entries.hasMoreElements(); ) {
       ZipArchiveEntry entry = entries.nextElement();
-      if (!isInsideZipDirectory(entry)) {
-        filenamesBuilder.add(entry.getName());
+      String entryName = entry.getName();
+      // check if the first entry is a directory with the name of the zip file
+      if (entry.isDirectory() && zipFileName.equals(entryName.replaceFirst("/", ""))) {
+        isMacZip = true;
+      }
+      Pair<String, Boolean> pair = isInsideZipDirectory(entry, macDirectory, entryName);
+      if (!pair.getLeft().isBlank() && !pair.getLeft().equals(".DS_Store") && !pair.getRight()) {
+        filenamesBuilder.add(pair.getLeft());
       }
     }
     filenames = filenamesBuilder.build();
   }
 
-  static boolean isInsideZipDirectory(ZipArchiveEntry entry) {
+  Pair<String, Boolean> isInsideZipDirectory(
+      ZipArchiveEntry entry, String macDirectory, String entryName) {
     // We do not use File.separator because the .zip file specification states:
     // All slashes MUST be forward slashes '/' as opposed to backwards slashes '\' for compatibility
     // with Amiga and
     // UNIX file systems etc.
     //
     // Directory names in end with '/'.
-    return entry.getName().contains("/");
+    if (isMacZip) {
+      entryName = entry.getName().replaceFirst(macDirectory, "");
+    }
+    Pair<String, Boolean> pair = Pair.of(entryName, entryName.contains("/"));
```

Comment:
To continue on the comment above, remove the Pair and replace with somethning like:
```
// Check for .DS_Store because ....
return entryName.isBlank() || entryName.contains("/") || entryName.equals(".DS_Store");
```

Reply:
good point, just add the comments in code.